src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,63 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { Filter } from '../utils';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filterBy: Filter;
+  query: string,
+  setFilterBy: (e: Filter) => void,
+  setQuery: (e: string) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter = ({
+  filterBy,
+  query,
+  setFilterBy,
+  setQuery,
+}: Props) => {
+  // const [hide, setHide] = useState<boolean>(true);
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filterBy}
+            onChange={(e) => setFilterBy(e.target.value as Filter)}
```

Comment:
typescript warns you it can be null - please handle it correctly instead of just binding it as Filter